zerver/lib/utils.py

```diff
@@ -33,3 +33,20 @@ def process_list_in_batches(
             break
         process_batch(items)
         offset += chunk_size
+
+
+def remove_single_newlines(text: str) -> str:
+    text = text.strip("\n")
+    updated_text = ""
+    prev_line_empty = False
+    for line in text.split("\n"):
+        line = line.strip()
+        if line:
+            if prev_line_empty:
+                updated_text = updated_text.rstrip()
+                updated_text += "\n\n"
+            updated_text += line + " "
+            prev_line_empty = False
+        else:
+            prev_line_empty = True
+    return updated_text.rstrip()
```

Comment:
This might be possible to do with a simple regular expression replacing `[^\n]\n[^\n]` with ` `? Not sure, and not sure it's worth extra effort to do so.

But in any case, it is definitely worth having a test for this function.

Reply:
Yeah using regex is a cleaner approach:
```
def remove_single_newlines(content: str) -> str:
    content = content.strip("\n")
    return re.sub(r"(?<!\n)\n(?!\n)", " ", content)
```
Added tests.